src/client/actions/MessageReactionAdd.js

```diff
@@ -18,8 +18,14 @@ class MessageReactionAdd extends Action {
   handle(data) {
     if (!data.emoji) return false;
 
-    const user = this.getUserFromMember(data);
-    if (!user) return false;
+    const guild = this.client.guilds.cache.get(data.guild_id);
+    let member = null;
+    if (guild) member = this.getMember({ user: data.member.user }, guild);
+    else member = this.getUserFromMember(data);
```

Comment:
This results in the event emitting with unreliable types - developers would now need to check if its emitting a User or a GuildMember. This is less desirable imo.

The event should probably emit a third optional `member` in addition to the guaranteed user.

Reply:
They can do that by checking if the event was triggered in or outside of DMs since it only emits user when it's not in a guild. I  don't think it would be too hard to do honestly.